lib/statsd/instrument/aggregator.rb

```diff
@@ -0,0 +1,260 @@
+# frozen_string_literal: true
+
+module StatsD
+  module Instrument
+    class AggregationValue
+      attr_accessor :value
+
+      attr_reader :name, :type, :tags, :no_prefix
+
+      # @param name [String] The name of the metric.
+      # @param type [Symbol] The type of the metric.
+      # @param value [Integer, Array<Float>] The value of the metric.
+      def initialize(name, type, value, tags, no_prefix)
+        @name = name
+        @type = type
+        @value = value
+        @tags = tags
+        @no_prefix = no_prefix
+      end
+    end
+
+    class Aggregator
+      CONST_SAMPLE_RATE = 1.0
```

Comment:
I don't understand why you need to disregard sampling. You could perfectly handle it AFAICT.

Reply:
Yeah, I kind of agree as well, on the other hand sampling is probabilistic. There is a discussion about this here: https://github.com/Shopify/statsd-instrument/pull/366#discussion_r1635323269. More notably, datadog uses the same strategies on their Ruby and Golang clients.